python/cudf/cudf/core/column/interval.py

```diff
@@ -80,12 +85,38 @@ def copy(self, deep=True):
         struct_copy = super().copy(deep=deep)
         return IntervalColumn(
             size=struct_copy.size,
-            dtype=cudf.core.dtypes.IntervalDtype(
-                struct_copy.dtype.fields["left"], closed
-            ),
+            dtype=IntervalDtype(struct_copy.dtype.fields["left"], closed),
             mask=struct_copy.base_mask,
             offset=struct_copy.offset,
             null_count=struct_copy.null_count,
             children=struct_copy.base_children,
             closed=closed,
         )
+
+    def as_interval_column(self, dtype, **kwargs):
+        if is_interval_dtype(dtype):
+            try:
+                return IntervalColumn(
+                    size=self.size,
+                    dtype=dtype,
+                    mask=self.mask,
+                    offset=self.offset,
+                    null_count=self.null_count,
+                    children=self.children,
+                    closed=dtype.closed
+                )
+            except: 
+                #this block is for the case where the dtype does not have 
+                #a closed attribute and therefore throws an error
```

Comment:
When would an interval dtype not have a `closed` attribute?

Reply:
An example is when we pass in a `list` of pandas intervals to IntervalIndex e.g. `cudf.IntervalIndex([pd.Interval(30, 50, closed='right')]), closed = 'right')` 
The data passes through and lands at 
```
else:
  data = column.as_column(
                  data, dtype="interval" if dtype is None else dtype
              )
```
Which assigns it the string dtype `"interval"`. I'm not checking for a list of intervals but can do that 😅
If there aren't any other edge cases then we shouldn't need the try, except block.

Reply:
Also running tests reminded me that the other case this comes up is if a user is creating a series or dataframe and uses the string `"interval"` as the dtype. I removed the try block and used the following code to change it to a cudf Interval Dtype right away instead! Works nicely 😄 

```
if dtype == 'interval':
                dtype = IntervalDtype(self.dtype.fields["left"], self.closed)
```